### PR TITLE
Add TT_PRELOAD doc

### LINKED ISSUE #3268: TT_PRELOAD
**Related dev. issue(s):** https://github.com/tarantool/tarantool/issues/7714

**Product:** Tarantool
**Since:** 2.11
**Root document:** https://www.tarantool.io/en/doc/latest/book/admin/instance_config/
**SME:** @ Totktonada

# Details
A script file or a module may be specified in the `TT_PRELOAD`
environment variable to load before the main script. For example:

```shell
$ TT_PRELOAD=/path/to/foo.lua tarantool main.lua
```

The `foo.lua` script will be executed before `main.lua`.

```shell
$ TT_PRELOAD=foo.bar.baz tarantool main.lua
```

The `foo.bar.baz` module will be loaded before executing `main.lua`.

A value that ends with `.lua` is considered as a script file, otherwise
it is considered as a module name.

Several files or modules may be passed as a semicolon separated list:

```shell
$ TT_PRELOAD="/path/to/foo.lua;foo.bar.baz" tarantool main.lua
```

Don't forget quotes, a shell interprets a semicolon on its own.

Redundant (duplicated, leading, trailing) semicolons are ignored.

If something goes wrong at script/module execution, tarantool reports
the problem and exits.

A script receives a path to the file in `...`, a module gets the module
name. `arg` is the same as for the main script.
Requested by @ Totktonada in https://github.com/tarantool/tarantool/commit/435e5efe982025233977effa3dab27e729c57c8a.

### PR DESCRIPTION
**Pull request by p7nov**
Resolves #3268 

Deployment: https://docs.d.tarantool.io/en/doc/gh-3268-tt-preload/book/admin/instance_config/#preloading-lua-scripts-and-modules

### CODE REVIEW

File: doc/book/admin/instance_config.rst [hunk 1]
```diff
@@ -84,11 +82,51 @@ entire application business logic in it. We, however, do not recommend this,
 since it clutters the instance file and leads to unnecessary copy-paste when
 you need to run multiple instances of an application.
 
+.. _admin-tt-preload:
+
+Preloading Lua scripts and modules
```
Comment (Totktonada) [inline doc/book/admin/instance_config.rst:87]: I would add the following information. Quoted the documentation request.

> Don't forget quotes, a shell interprets a semicolon on its own.
>
> If something goes wrong at script/module execution, tarantool reports
the problem and exits.
>
> A script receives a path to the file in `...`, a module gets the module
name. `arg` is the same as for the main script.

**Review by Totktonada — APPROVED** [review]

File: doc/book/admin/instance_config.rst (hunk 1 above)
Comment (p7nov) [reply doc/book/admin/instance_config.rst:87]: Added these details.

File: doc/book/admin/instance_config.rst [hunk 2]
```diff
@@ -84,11 +82,83 @@ entire application business logic in it. We, however, do not recommend this,
...
+----------------------------------
+
+Tarantool supports loading and running chunks of Lua code before the loading instance file.
+To load or run Lua code immediately upon Tarantool startup, specify the ``TT_PRELOAD``
+environment variable. Its value can be either a path to a Lua script, or a Lua module name:
```
Comment (andreyaksenov) [inline doc/book/admin/instance_config.rst]:
```suggestion
environment variable. Its value can be either a path to a Lua script or a Lua module name:
```

File: doc/book/admin/instance_config.rst [hunk 3]
```diff
@@ -84,11 +82,83 @@ entire application business logic in it. We, however, do not recommend this,
...
+
+        $ TT_PRELOAD=preload.module tarantool main.lua
+
+    Tarantool loads the ``preload.module`` code into the interpreter and
+    starts running ``main.lua`` as if its first statement was ``require('preload.module')``.
+
+    .. warning::
+
+        ``TT_PRELOAD`` values that end with ``.lua`` are considered scripts,
+        so avoid modules names with this ending.
```
Comment (andreyaksenov) [inline doc/book/admin/instance_config.rst]:
```suggestion
        so avoid module names with this ending.
```

File: doc/book/admin/instance_config.rst [hunk 4]
```diff
@@ -84,11 +82,83 @@ entire application business logic in it. We, however, do not recommend this,
...
+
+Preloading Lua scripts and modules
+----------------------------------
+
+Tarantool supports loading and running chunks of Lua code before the loading instance file.
+To load or run Lua code immediately upon Tarantool startup, specify the ``TT_PRELOAD``
+environment variable. Its value can be either a path to a Lua script, or a Lua module name:
+
+*   To run the Lua script ``script.lua`` from the ``preload/path/`` directory inside
+    the working directory in Tarantool before executing ``main.lua``:
```
Comment (andreyaksenov) [inline doc/book/admin/instance_config.rst]: Maybe it's a matter of taste but I'd make this sentence full: `To run ..., set TT_PRELOAD as follows:`. I couldn't read it from the first attempt and looked for continuation :)

**Review by andreyaksenov — APPROVED** [review]

**Pull request merged**